openfl/utilities/verifdataset/verifiable_dataset.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+""" This module contains the VerifiableDataset class. """
+
+import json
+from enum import Enum
+from hashlib import sha384
+from pathlib import Path
+from typing import List
+
+from openfl.utilities.verifdataset.data_source import DataSource, DataSourceType
+from openfl.utilities.verifdataset.local_data_source import LocalDataSource
+
+
+class DatasetFormat(Enum):
+    """Enum for the different dataset commitment formats."""
+
+    VERBOSE = "verbose_dataset"
+    CONCISE = "concise_dataset"
+
+
+def filter_non_serializable(obj):
+    """Filter out methods and non-serializable objects."""
+    serializable_dict = {}
+    for key, val in obj.__dict__.items():
+        if callable(val):
+            continue  # Skip methods
+        if isinstance(val, Enum):
+            val = val.value  # Convert Enum to its value
+        elif isinstance(val, Path):
+            val = str(val)  # Convert Path to string
+        serializable_dict[key] = val
+    return serializable_dict
+
+
+class VerifiableDataset:
+    """
+    This class represents a data set that can be verified.
+    It is containing multiple data sources and contains methods to compute the commitment,
+    validation, serialization and deserialization
+    """
+
+    def __init__(
+        self,
+        data_sources: List[DataSource],
+        base_dir: Path,
+        label: str,
+        metadata,
+        dataset_format: DatasetFormat,
+        valid_hash=None,
```

Comment:
I suggest simply `hash=None`

Reply:
yeah. some linter I tried didn't like it but I see that ours is good with that so I renamed to `hash`. thanks